enhancements/microshift/disabling-storage-components.md

```diff
@@ -0,0 +1,243 @@
+  but do not require volume snapshotting.
+- A user is operating MicroShift on a small-form factor machine, is reducing resource consumption wherever possible, and
+does not have a requirement for persistent storage.
+
+### Goals
+
+- Enhance the MicroShift config API to support selective deployment of LVMS and CSI Snapshotting.
+- Do not make backwards-incompatible changes to the MicroShift config API.
+- Do not endanger or make inaccessible persistent data on upgraded clusters.
+
+### Non-Goals
+
+- Provide a generalized framework to support potential future alternatives to LVMS.
+- Generically enable installing CSI components without enabling LVMS.
+
+## Proposal
+
+### Workflow Description
+
+**_Installation with LVMS and Snapshotting (Default)_**
+
+1. User installs MicroShift. A MicroShift and LVMD config are not provided.
+2. MicroShift starts and detects no config and falls back to default MicroShift configuration (LVMS and CSI on by default).
+3. MicroShift checks to determine if host is compatible with LVMS. If it is not, an error is logged, 
+LVMS and CSI installation is skipped and install continues. (This it the current behavior in MicroShift)
+4. MicroShift attempts to dynamically create generate the LVMD config. If it cannot, an error is logged, LVMS and CSI
+installation is skipped and install continues.
+5. If all checks pass, LVMS and CSI are deployed.
+
+**_Installation without LVMS and Snapshotting_**
+
+1. User installs MicroShift.
+2. User specifies a MicroShift config sets fields to disable LVMS and CSI Snapshots.
+3. MicroShift starts and detects the provided config.
+4. LVMS and CSI snapshot components are not deployed.
+
+**_Post-Start Installation, LVMS only_**
+
+1. User has already installed and started MicroShift service.
+2. User later determines there is a requirement for persistent storage, but not snapshotting.
+3. User edits the MicroShift config, enabling LVMS only.
+4. User restarts MicroShift service.
+5. MicroShift starts and detects the provided config.
+6. MicroShift performs LVMS startup checks.
+7. If all checks pass, LVMS is deployed. Else if checks fail, an error is logged, LVMS deployment is skipped, 
+and startup continues.
+
+**_Complete Uninstallation, with Data removal_**
+
+> NOTE: Installation is not reversible. Deleting LVMS or the CSI snapshotter while there are still storage volumes risks
+orphaning volumes. It is the user's responsibility to manually destroy data volumes and/or snapshots before
+uninstalling.
+
+1. User has already deployed a cluster with LVMS and CSI Snapshotting installed.
+2. User has deployed cluster workloads with persistent storage.
+3. User decides that LVMS and snapshotting are no longer needed.
+4. User edits `/etc/microshift/config.yaml`, setting `.storage.driver: none`. 
+5. User takes actions to back up, wipe, or otherwise ensure data will not be irretrievable.
```

Comment:
"ensure data will not be irretrievable" - has a double-negative, and therefore is semantically equivalent to "ensure that data will be retrievable". I think we want here is "User takes steps to back up, and then erase, or otherwise ensure that data cannot be recovered."